boefjes/boefjes/katalogus/dependencies/plugins.py

```diff
@@ -75,7 +75,17 @@ def by_plugin_id(self, plugin_id: str, organisation_id: str) -> PluginType:
         raise KeyError(f"Plugin {plugin_id} not found for {organisation_id}")
 
     def by_plugin_ids(self, plugin_ids: list[str], organisation_id: str) -> list[PluginType]:
-        return [self.by_plugin_id(plugin_id, organisation_id) for plugin_id in plugin_ids]
+        all_plugins = self.get_all(organisation_id)
+        plugin_dict: dict[str, PluginType] = {plugin.id: plugin for plugin in all_plugins}
```

Comment:
Although this works, you don't need to create this plugins map. You can just iterate over `all_plugins` and do the ID checks. This will save another loop and dictionary creation.